opentelemetry-stdout/src/trace/exporter.rs

```diff
@@ -22,106 +20,136 @@ impl fmt::Debug for SpanExporter {
-    /// ```
-    pub fn with_writer(mut self, writer: impl Write + Send + Sync + 'static) -> Self {
-        self.writer = Some(Box::new(writer));
-        self
-    }
+        let mut print_header = true;
+        for kv in span.attributes.iter() {
+            if print_header {
+                println!("\t Attributes:");
+                print_header = false;
+            }
+            println!("\t\t ->  {}: {:?}", kv.key, kv.value);
+        }
 
-    /// Set the encoder that this exporter will use
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use opentelemetry_stdout::SpanExporterBuilder;
-    ///
-    /// let exporter = SpanExporterBuilder::default()
-    ///     // Can be any function that can write formatted data
-    ///     // serde ecosystem crates for example provide such functions
-    ///     .with_encoder(|writer, data| Ok(serde_json::to_writer_pretty(writer, &data).unwrap()))
-    ///     .build();
-    /// ```
-    pub fn with_encoder(
-        mut self,
-        writer: impl Fn(&mut dyn Write, SpanData) -> TraceResult<()> + Send + Sync + 'static,
-    ) -> Self {
-        self.encoder = Some(Box::new(writer));
-        self
-    }
+        print_header = true;
+        for event in span.events.iter() {
+            if print_header {
+                println!("\t Events:");
+                print_header = false;
+            }
+            println!("\t\t Name: {:?}", event.name);
+            let datetime: DateTime<Utc> = event.timestamp.into();
+            println!(
+                "\t\t Timestamp: {}",
+                datetime.format("%Y-%m-%d %H:%M:%S%.6f")
+            );
+            let mut print_header_event_attributes = true;
+            for kv in event.attributes.iter() {
+                if print_header_event_attributes {
+                    println!("\t\t Attributes:");
+                    print_header_event_attributes = false;
+                }
+                println!("\t\t\t ->  {}: {:?}", kv.key, kv.value);
+            }
+        }
 
-    /// Create a span exporter with the current configuration
-    pub fn build(self) -> SpanExporter {
-        SpanExporter {
-            writer: Some(self.writer.unwrap_or_else(|| Box::new(stdout()))),
-            resource: Resource::empty(),
-            encoder: self.encoder.unwrap_or_else(|| {
-                Box::new(|writer, spans| {
-                    serde_json::to_writer(writer, &spans)
-                        .map_err(|err| TraceError::Other(Box::new(err)))
-                })
-            }),
+        print_header = true;
+        for link in span.links.iter() {
+            if print_header {
+                println!("\t Links:");
+                print_header = false;
+            }
+            println!("\t\t TraceId: {:?}", link.span_context.trace_id());
+            println!("\t\t SpanId: {:?}", link.span_context.span_id());
+            println!("\t\t Attributes:");
```

Comment:
Can we do 
```
span.links.iter()
    .enumerate()
    .for_each(|(index, link)| {
        if index == 0 {
            println!("\t Links:");
        }

        println!("\t\t TraceId: {:?}", link.span_context.trace_id());
        println!("\t\t SpanId: {:?}", link.span_context.span_id());
        ...
   })
```
no need for the `print_header` variable, if our intention here is to print once

Reply:
Thanks! Yea that looks much better!
I believe next version of clippy will be able to detect/suggest these improvements automatically! (it already suggested fixes when I maintained "i" for counting iterations!)